packager/media/test/test_data_util.cc

```diff
@@ -4,41 +4,42 @@
 
 #include "packager/media/test/test_data_util.h"
 
-#include "packager/base/files/file_util.h"
-#include "packager/base/logging.h"
-#include "packager/base/path_service.h"
+#include "glog/logging.h"
 
 namespace shaka {
 namespace media {
 
-base::FilePath GetTestDataFilePath(const std::string& name) {
-  base::FilePath file_path;
-  CHECK(PathService::Get(base::DIR_SOURCE_ROOT, &file_path));
-
-  file_path = file_path.Append(FILE_PATH_LITERAL("packager"))
-                  .Append(FILE_PATH_LITERAL("media"))
-                  .Append(FILE_PATH_LITERAL("test"))
-                  .Append(FILE_PATH_LITERAL("data"))
-                  .AppendASCII(name);
-  return file_path;
+// Returns a file path for a file in the media/test/data directory.
+std::filesystem::path GetTestDataFilePath(const std::string& name) {
+  std::filesystem::path header_path(__FILE__);
+  return header_path.parent_path() / "data" / name;
 }
 
-base::FilePath GetAppTestDataFilePath(const std::string& name) {
-  base::FilePath file_path;
-  CHECK(PathService::Get(base::DIR_SOURCE_ROOT, &file_path));
-
-  file_path = file_path.Append(FILE_PATH_LITERAL("packager"))
-                  .Append(FILE_PATH_LITERAL("app"))
-                  .Append(FILE_PATH_LITERAL("test"))
-                  .Append(FILE_PATH_LITERAL("testdata"))
-                  .AppendASCII(name);
-  return file_path;
+// Returns a file path for a file in the media/app/test/testdata directory.
+std::filesystem::path GetAppTestDataFilePath(const std::string& name) {
+  std::filesystem::path header_path(__FILE__);
+  return header_path.parent_path().parent_path() / "app" / "test" / "testdata" /
+         name;
 }
 
+// Reads a test file from media/test/data directory and returns its content.
 std::vector<uint8_t> ReadTestDataFile(const std::string& name) {
-  std::string buffer;
-  CHECK(base::ReadFileToString(GetTestDataFilePath(name), &buffer));
-  return std::vector<uint8_t>(buffer.begin(), buffer.end());
+  std::filesystem::path path = GetTestDataFilePath(name);
+
+  std::vector<uint8_t> data;
+  data.resize(std::filesystem::file_size(path));
+
+  FILE* f = fopen(path.string().c_str(), "rb");
+  if (!f) {
+    LOG(FATAL) << "Failed to read test data from " << path;
+    return std::vector<uint8_t>();
```

Comment:
if you'd move `data.resize` to line 37 you could return here `data` as it would be already created and empty
still resizing makes sense in line 37 as we are sure that file open was successful

Reply:
Moved.  Thanks!